### PR TITLE
Grammar

### PR DESCRIPTION
**Pull request by jsoref**
* dashed compound words
* articles
* pronoun agreement - which/that
* count agreement

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # node-problem-detector
 
-[![Build Status](https://travis-ci.org/kubernetes/node-problem-detector.svg?branch=master)](https://travis-ci.org/kubernetes/node-problem-detector)  [![Go Report Card](https://goreportcard.com/badge/github.com/kubernetes/node-problem-detector)](https://goreportcard.com/report/github.com/kubernetes/node-problem-detector)
+[![Build Status](https://travis-ci.or,-detector.svg?branch=master)](https://travis-ci.org/kubernetes/node-problem-detector)  [![Go Report Card](https://goreportcard.com/badge/github.com/kubernetes/node-problem-detector)](https://goreportcard.com/report/github.com/kubernetes/node-problem-detector)
```
Comment (xueweiz) [inline README.md:3]: Is this a typo?

File: README.md [hunk 2]
```diff
@@ -17,17 +17,17 @@ enabled by default in the GCE cluster.
 There are tons of node problems that could possibly affect the pods running on the
 node, such as:
 * Infrastructure daemon issues: ntp service down;
-* Hardware issues: Bad cpu, memory or disk;
+* Hardware issues: Bad CPU, memory, or disk;
```
Comment (xueweiz) [inline README.md:20]: I think this is a bit obsessive, right?
For situation like this where both ways are correct [1], I'd prefer to just leave it as it. There is not much benefit of switching around.

[1] https://www.grammarly.com/blog/comma-before-and/

File: README.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-[![Build Status](https://travis-ci.org/kubernetes/node-problem-detector.svg?branch=master)](https://travis-ci.org/kubernetes/node-problem-detector)  [![Go Report Card](https://goreportcard.com/badge/github.com/kubernetes/node-problem-detector)](https://goreportcard.com/report/github.com/kubernetes/node-problem-detector)
+[![Build Status](https://travis-ci.or,-detector.svg?branch=master)](https://travis-ci.org/kubernetes/node-problem-detector)  [![Go Report Card](https://goreportcard.com/badge/github.com/kubernetes/node-problem-detector)](https://goreportcard.com/report/github.com/kubernetes/node-problem-detector)
 
 node-problem-detector aims to make various node problems visible to the upstream
-layers in cluster management stack.
-It is a daemon which runs on each node, detects node
-problems and reports them to apiserver.
+layers in the cluster management stack.
+It is a daemon that runs on each node, detects node
+problems, and reports them to apiserver.
```
Comment (xueweiz) [inline README.md:8]: I also think this is a bit obsessive. Whatever that's not breaking, there is no point fixing it.

File: README.md [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
 
-[![Build Status](https://travis-ci.org/kubernetes/node-problem-detector.svg?branch=master)](https://travis-ci.org/kubernetes/node-problem-detector)  [![Go Report Card](https://goreportcard.com/badge/github.com/kubernetes/node-problem-detector)](https://goreportcard.com/report/github.com/kubernetes/node-problem-detector)
+[![Build Status](https://travis-ci.or,-detector.svg?branch=master)](https://travis-ci.org/kubernetes/node-problem-detector)  [![Go Report Card](https://goreportcard.com/badge/github.com/kubernetes/node-problem-detector)](https://goreportcard.com/report/github.com/kubernetes/node-problem-detector)
 
 node-problem-detector aims to make various node problems visible to the upstream
-layers in cluster management stack.
-It is a daemon which runs on each node, detects node
-problems and reports them to apiserver.
+layers in the cluster management stack.
+It is a daemon that runs on each node, detects node
```
Comment (xueweiz) [inline README.md:7]: Could you please explain why is "that" better than "which" in this situation?

File: README.md [hunk 5]
```diff
@@ -17,17 +17,17 @@ enabled by default in the GCE cluster.
...
 node, such as:
 * Infrastructure daemon issues: ntp service down;
-* Hardware issues: Bad cpu, memory or disk;
+* Hardware issues: Bad CPU, memory, or disk;
 * Kernel issues: Kernel deadlock, corrupted file system;
 * Container runtime issues: Unresponsive runtime daemon;
 * ...
 
-Currently these problems are invisible to the upstream layers in cluster management
+Currently, these problems are invisible to the upstream layers in the cluster management
```
Comment (xueweiz) [inline README.md:25]: Why is this necessary?

File: README.md [hunk 6]
```diff
@@ -45,7 +45,7 @@ A problem daemon is a sub-daemon of node-problem-detector. It monitors a specifi
 kind of node problems and reports them to node-problem-detector.
 
 A problem daemon could be:
-* A tiny daemon designed for dedicated usecase of Kubernetes.
+* A tiny daemon designed for the dedicated Kubernetes use-case.
```
Comment (xueweiz) [inline README.md:48]: I'd prefer:
* A tiny daemon designed for dedicated Kubernetes use-cases.

File: README.md [hunk 7]
```diff
@@ -54,22 +54,22 @@ different containers, and compose them with pod specification.
 
 Each category of problem daemon can be disabled at compilation time by setting
 corresponding build tags. If they are disabled at compilation time, then all their
-build dependencies, global variables and background goroutines will be trimmed out
+build dependencies, global variables, and background goroutines will be trimmed out
```
Comment (xueweiz) [inline README.md:57]: ditto

File: README.md (hunk 1 above)
Comment (jsoref) [reply README.md:3]: eep, that's definitely keyboard/trackpad selection damage.

File: README.md (hunk 2 above)
Comment (jsoref) [reply README.md:20]: Sometimes there's a certain amount of ambiguity which is fixed by using Oxford commas.

I'm not terribly attached to them. And given the damage above, if you're ok w/ a force push, I'm happy to wipe them out.

File: README.md (hunk 4 above)
Comment (jsoref) [reply README.md:7]: https://www.grammarly.com/blog/which-vs-that/

File: README.md (hunk 6 above)
Comment (jsoref) [reply README.md:48]: (Thanks, I actually struggled with trying to come up w/ a good way to write this.)

File: README.md (hunk 1 above)
Comment (jsoref) [reply README.md:3]: (It's also evidence that I'm a human... -- I'm actually fixing similar damage in someone else's repository...)

File: README.md (hunk 5 above)
Comment (jsoref) [reply README.md:25]: https://www.grammarly.com/blog/commas-after-introductory-words/

**Comment by jsoref** [comment]
@xueweiz I've addressed most of your comments and left references for the others.

**Comment by jsoref** [comment]
@xueweiz?
